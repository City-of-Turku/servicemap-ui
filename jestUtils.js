import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });