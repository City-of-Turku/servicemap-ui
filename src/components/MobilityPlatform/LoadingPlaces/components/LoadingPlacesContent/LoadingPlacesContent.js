import PropTypes from 'prop-types';
import React from 'react';
import SingleValueText from '../../../SingleValueText';
import TextComponent from '../../../TextComponent';

const LoadingPlacesContent = ({ classes, item }) => {
  const loadingPlaceName = {
    fi: item.name_fi,
    en: item.name_en,
    sv: item.name_sv,
  };

  const loadingPlaceAddress = {
    fi: item.address_fi,
    en: item.address_en,
    sv: item.address_sv,
  };

  const loadingPlaceInfo = (
    <div className={classes.container}>
      <div className={classes.headerContainer}>
        <TextComponent textObj={loadingPlaceName} isTitle />
      </div>
      <div className={classes.textContainer}>
        {item.address_fi !== '' ? <SingleValueText messageId="mobilityPlatform.content.address" textObj={loadingPlaceAddress} /> : null}
        <TextComponent textObj={item.extra.lastauspiste} />
        <TextComponent textObj={item.extra.Saavutettavuus} />
        <TextComponent textObj={item.extra.rajoitustyyppi} />
      </div>
    </div>
  );

  return (
    <div className={classes.container}>
      {loadingPlaceInfo}
    </div>
  );
};

LoadingPlacesContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  item: PropTypes.objectOf(PropTypes.any),
};

LoadingPlacesContent.defaultProps = {
  item: {},
};

export default LoadingPlacesContent;
