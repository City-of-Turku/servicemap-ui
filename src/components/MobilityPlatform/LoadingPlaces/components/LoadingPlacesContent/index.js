import { withStyles } from '@mui/styles';
import LoadingPlacesContent from './LoadingPlacesContent';
import styles from './styles';

export default withStyles(styles)(LoadingPlacesContent);
