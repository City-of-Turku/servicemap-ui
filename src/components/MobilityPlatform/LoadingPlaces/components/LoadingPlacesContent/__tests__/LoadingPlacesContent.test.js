// Link.react.test.js
import React from 'react';
import LoadingPlacesContent from '../index';
import { initialState } from '../../../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  item: {
    name_fi: 'Testi',
    name_en: 'Test',
    name_sv: 'Test',
    address_fi: 'Testikatu',
    address_en: 'Test street',
    address_sv: 'Test gata',
    extra: {
      lastauspiste: {
        fi: 'Testi',
        en: 'Test',
        sv: 'Test',
      },
      rajoitustyyppi: {
        fi: 'Testitieto',
        en: 'Test info',
        sv: 'Test info',
      },
      Saavutettavuus: {
        fi: 'Testitieto',
        en: 'Test info',
        sv: 'Test info',
      },
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<LoadingPlacesContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<LoadingPlacesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<LoadingPlacesContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.item.name_fi);
    expect(p[1].textContent).toContain(`Osoite: ${mockProps.item.address_fi}`);
    expect(p[2].textContent).toContain(mockProps.item.extra.lastauspiste.fi);
    expect(p[3].textContent).toContain(mockProps.item.extra.Saavutettavuus.fi);
    expect(p[4].textContent).toContain(mockProps.item.extra.rajoitustyyppi.fi);
  });
});
