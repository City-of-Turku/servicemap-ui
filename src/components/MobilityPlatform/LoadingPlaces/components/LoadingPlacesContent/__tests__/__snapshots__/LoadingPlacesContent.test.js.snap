// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingPlacesContent /> should match snapshot 1`] = `
<div>
  <div
    class="LoadingPlacesContent-container-1"
  >
    <div
      class="LoadingPlacesContent-container-1"
    >
      <div
        class="LoadingPlacesContent-headerContainer-2"
      >
        <div
          class="css-qpxa1n"
        >
          <p
            class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
          >
            Testi
          </p>
        </div>
      </div>
      <div
        class="LoadingPlacesContent-textContainer-3"
      >
        <div
          class="css-qpxa1n"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Osoite: Testikatu
          </p>
        </div>
        <div
          class="css-qpxa1n"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Testi
          </p>
        </div>
        <div
          class="css-qpxa1n"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Testitieto
          </p>
        </div>
        <div
          class="css-qpxa1n"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Testitieto
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
